File: app/src/main/java/com/example/blockgraming19/PrintStatement.java
package com.example.blockgraming19;
public final class PrintStatement implements Statement {
    
    private final Expression expression;

    public PrintStatement(Expression expression) {
        this.expression = expression;
    }

    @Override
    public void execute() {
        System.out.print(expression.eval());
    }

    @Override
    public String toString() {
        return "print " + expression;
    }
}

File: app/src/main/java/com/example/blockgraming19/PrintStatement.kt
package com.example.blockgraming19

import android.widget.TextView

class PrintStatement(private val expression: Expression, private var textView: TextView) : Statement {
    override fun execute() {
        print(expression.eval())
        textView.setText(textView.text.toString() + (expression.eval()).toString() + "\n")
    }

    override fun toString(): String {
        return "print $expression"
    }
}